File: examples/docs-code-snippets/src/main/kotlin/examples/kotlin/recipes/RecipeWithRequiredEvents.kt
package examples.kotlin.recipes

import com.ing.baker.recipe.kotlindsl.ExperimentalDsl
import com.ing.baker.recipe.kotlindsl.recipe
import examples.kotlin.events.FraudCheckCompleted
import examples.kotlin.events.PaymentReceived
import examples.kotlin.interactions.ShipOrder

@ExperimentalDsl
object RecipeWithRequiredEvents {
    val recipe = recipe("example") {
        interaction<ShipOrder> {
            requiredEvents {
                event<FraudCheckCompleted>()
            }
            requiredOneOfEvents {
                event<PaymentReceived>()
                event(name = "UsedCouponCode")
            }
        }
    }
}

File: examples/docs-code-snippets/src/main/scala/examples/scala/recipes/RecipeWithRequiredEvents.scala
package examples.scala.recipes

import com.ing.baker.recipe.scaladsl.{Event, Recipe}
import examples.scala.events.{FraudCheckCompleted, PaymentReceived}
import examples.scala.interactions.ShipOrder

object RecipeWithRequiredEvents {
  val recipe: Recipe = Recipe("example")
    .withInteraction(
      ShipOrder.interaction
        .withRequiredEvent(
          FraudCheckCompleted.event
        )
        .withRequiredOneOfEvents(
          PaymentReceived.event,
          Event("UsedCouponCode")
        )
    )
}